lib/rspec/mocks/any_instance/recorder.rb

```diff
@@ -190,9 +193,9 @@ def restore_original_method!(method_name)
           return unless @klass.instance_method(method_name).owner == @klass
 
           alias_method_name = build_alias_method_name(method_name)
-          @klass.class_exec do
+          @klass.class_exec(@own_methods) do |own_methods|
             remove_method method_name
-            alias_method method_name, alias_method_name
+            alias_method method_name, alias_method_name if own_methods.include?(method_name.to_sym)
```

Comment:
And also, this one will be lost on a force push...

> Thanks for taking a stab at this, @jrafanie. I'm a bit concerned about the performance implications of your solution here -- mapping over all methods (e.g. map(&:to_sym)) is a linear operation that we'd like to avoid (particularly because the sorts of legacy systems where users use any_instance tend to be ones where there objects have very large interfaces). Likewise, the include? check is a linear operation as well.
Did an owner check not work? That route seems simpler and likely to perform better.